### ISSUE TITLE
`poetry lock` should overwrite broken `poetry.lock` file, not error out.

### ISSUE THREAD

**Issue by dimaqq** [python-poetry/poetry#1196]
```
… ~/my_project> poetry shell

[EmptyKeyError]
Empty key at line 36 col 0

shell

```

The actual reason is that I have a merge conflict in `pyproject.toml`.
I wish `poetry` told me that! 🙏 

Also, if the merge conflict is in `poetry.lock` the best action to resolve it is seemingly to run `poetry lock` (assuming project is close to `HEAD` and not some old branch merged with another old branch).

(poetry versions tested: `1.0.0a2` `1.0.0a4` `0.12.16`)

**Comment by dimaqq** [comment]
```
… ~/myproj> git rebase origin/HEAD
...
M	poetry.lock
Falling back to patching base and 3-way merge...
Auto-merging poetry.lock
CONFLICT (content): Merge conflict in poetry.lock
error: Failed to merge in the changes.
...
… ~/myproj> poetry lock

[EmptyKeyError]
Empty key at line 1896 col 0
```

Still a problem with `poetry==1.0.0b2`

**Comment by eikevons** [comment]
Would a `--force` flag help here, that ignores any existing content in `poetry.lock`. That way we avoid silently overwriting inconsistent lock states.

**Comment by sinoroc** [comment]
@dimaqq  Does the issue still exist in newer poetry versions (>=1.1)?

**Comment by eikevons** [comment]
I can confirm that a broken `poetry.lock` still makes `poetry lock` to throw an error.

In a fresh Python 3.8 with poetry 1.1.4 I get (I broke `poetry.lock` by emptying all content)
```
 poetry lock

  NonExistentKey

  'Key "metadata" does not exist.'

  at ~/.poetry/lib/poetry/_vendor/py3.8/tomlkit/container.py:576 in __getitem__
      572│             key = Key(key)
      573│ 
      574│         idx = self._map.get(key, None)
      575│         if idx is None:
    → 576│             raise NonExistentKey(key)
      577│ 
      578│         if isinstance(idx, tuple):
      579│             # The item we are getting is an out of order table
      580│             # so we need a proxy to retrieve the proper objects
```

Steps to reproduce with `docker`:
- create `test.sh`:
  ```sh
  #!/bin/sh

  set -xe

  apt-get update -q
  apt-get install --yes -q curl
  curl -sSL https://raw.githubusercontent.com/python-poetry/poetry/master/get-poetry.py | python
  . "$HOME/.poetry/env"
  mkdir -p /app
  cd /app
  poetry init -n -q
  poetry add -n -q requests
  echo -n > poetry.lock
  poetry lock
  ```
- run `test.sh`: `docker run --rm -ti --volume "$PWD/test.sh:/test.sh:ro" python:3.8-slim sh /test.sh`

**Comment by phillipuniverse** [comment]
Confirmed still an issue in 1.1.6. Looks like the only fix is `rm poetry.lock && poetry lock`.

The problem with that is that this will implicitly update dependencies, since you removed `poetry.lock` the `--no-update` doesn't actually do anything.

**Comment by kkirsche** [comment]
If there is concerns about the scope of this type of change, potentially an alternative is to restrict what cases this occurs in? For example, automate fixing merge conflicts potentially as `yarn` does?

**Comment by passionsfrucht** [comment]
Not really a solution, but a tip, or how I integrate it in my workflow regarding `--no-update`. It usually happens when I do a rebase, thus to hinder it to do an unintended upgrade:

```shell
$ git checkout origin/master -- poetry.lock
??? updated 1 file ???
$ poetry lock --no-update
```

I'm quite positive that it can be done in a more elegant way using `--ours` or some `.gitattributes` magic with a check if the lock file needs an update, but this way it is good enough for me right now.

So maybe it can be handled as a documentation issue?

Disclaimer: I just looked it up in my command history. And wrote the results how I remembered it, I may have got a detail wrong.
